src/modules/tests/test_data/test-data.reducer.ts

```diff
@@ -0,0 +1,31 @@
+import { TestData } from '@dvsa/mes-test-schema/categories/B';
+import * as testDataActions from './test-data.actions';
+import { createFeatureSelector } from '@ngrx/store';
+
+export const initialState: TestData = {
+  dangerousFaults: {},
+  drivingFaults: {},
+  manoeuvres: {},
+  seriousFaults: {},
+  testRequirements: {},
+};
+
+export function testDataReducer(
+  state = initialState,
+  action: testDataActions.Types,
+): TestData {
+  switch (action.type) {
+    case testDataActions.ADD_DRIVING_FAULT:
+      return {
+        ...state,
+        drivingFaults: Object.assign({}, state.drivingFaults, {
```

Comment:
Can we just use spread again here?
```
drivingFaults:  {
  ...state.drivingFaults,
  [action.payload.competency] : action.payload.newFaultCount,
}
```
Feels like that might create a TS error though

Reply:
Seems to work fine with the change